lib/mainloop-threaded-worker.c

```diff
@@ -37,6 +37,25 @@ main_loop_threaded_worker_signal_startup_finished(MainLoopThreadedWorker *self,
   g_mutex_unlock(&self->lock);
 }
 
+static gboolean
+_thread_init(MainLoopThreadedWorker *self)
+{
+  main_loop_worker_thread_start(self->worker_type);
+
+  gboolean result = self->thread_init(self);
```

Comment:
I think we should allow unimplemented `thread_init`/`thread_deinit()` methods.
(request_exit can be implemented with atomic variables, or the resource needed for this operation might be something that was allocated in the main thread at syslog-ng startup)